WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -63,6 +63,92 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
     ///
     private let variationDisplayMode: VariationDisplayMode?
 
+    /// Subscription settings extracted from product meta data for a Subscription-type Product, if any
+    ///
+    private(set) var productSubscriptionDetails: ProductSubscription?
+
+    /// Description of the subscription billing details for a Subscription-type Product
+    /// eg: "$60.00 / 2 months"
+    ///
+    var subscriptionBillingDetailsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        guard let subscriptionPrice = productSubscriptionDetails?.price,
+              let subscriptionInterval = productSubscriptionDetails?.periodInterval,
+              let subscriptionPeriod = productSubscriptionDetails?.period,
+              let formattedPrice = currencyFormatter.formatAmount(subscriptionPrice, with: currency) else {
+            return ""
+        }
+
+        let subscriptionFrequency = {
+            switch subscriptionInterval {
+            case "1":
+                return subscriptionPeriod.descriptionSingular
+            default:
+                return subscriptionPeriod.descriptionPlural
+            }
+        }
+        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
+                                                formattedPrice, subscriptionInterval, subscriptionFrequency())
+    }
+
+    /// Description of the subscription conditions for a Subscription-type Product
+    /// These are separate from each other, a subscription could have any, all, or none
+    /// eg: "$25.00 signup · 1 month free"
+    ///
+    var subscriptionConditionsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        // Signup fees
+        guard let signUpFee = productSubscriptionDetails?.signUpFee else {
+            return ""
+        }
```

Comment:
Just checking: Do we want to return an empty string if there is no signup fee? Are there other conditions we want to display in that case?

If yes, we should not use a `guard` check here. Please consider using `if let` instead.

Reply:
That's a good point, I wrote this bit under the assumption that while we treat these properties as optional in the app, the API seems to always return empty if do not exist, but I think that's the wrong assumption and you're right that we should treat these in a way that we do not want to return early at this point. Updated here: 241e1c86cd66f1622b58fccfe466551eb6d822f4